share/functions/__fish_svn_prompt.fish

```diff
@@ -0,0 +1,147 @@
+	# iterate over the different status types
+	for flag_type in $__fish_svn_prompt_flag_names
+		# resolve the name of the variable for the character representing the current status type
+		set -l flag_index (contains -i $flag_type $__fish_svn_prompt_flag_names)
+		# check to see if the status string for this column contains the character representing the current status type
+		if test (echo $flag_status_string | grep -c $__fish_svn_prompt_chars[$flag_index]) -eq 1
+			# if it does, then get the names of the variables for the display character and colour to format it with
+			set -l flag_var_display __fish_svn_prompt_char_{$flag_type}_display
+			set -l flag_var_color __fish_svn_prompt_char_{$flag_type}_color
+			# set the colour and print display character, then restore to default display colour
+			printf '%s%s%s' (set_color $$flag_var_color) $$flag_var_display (set_color normal)
+		end
+	end
+end
+
+
+function __fish_svn_prompt --description "Prompt function for svn"
+	# if svn isn't installed then don't do anything
+	if not command -s svn > /dev/null
+		return 1
+	end
+
+	# make sure that this is a svn repo
+	set -l checkout_info (command svn info ^/dev/null)
+	if [ $status -ne 0 ];
+		return
+	end
+
+	# get the current revision number
+	printf '(%s%s%s' (set_color $__fish_svn_prompt_color_revision) (__fish_print_svn_rev) (set_color normal)
+
+	# resolve the status of the checkout
+	# 1. perform `svn status`
+	# 2. remove extra lines that aren't necessary
+	# 3. cut the output down to the first 7 columns, as these contain the information needed
+	set -l svn_status_lines (command svn status | sed -e 's=^Summary of conflicts.*==' -e 's=^  Text conflicts.*==' -e 's=^  Property conflicts.*==' -e 's=^  Tree conflicts.*==' -e 's=.*incoming .* upon update.*==' | cut -c 1-7)
+
+	# track the last column to contain a status flag
+	set -l last_column 0
+
+	# iterate over the 7 columns of output (the 7 columns are defined on `svn help status`)
+	for col in (seq 7)
+		# get the output for a particular column
+		# 1. echo the whole status flag text
+		# 2. cut out the current column of characters
+		# 3. remove spaces and newline characters
+		set -l column_status (printf '%s\n' $svn_status_lines | cut -c $col | tr -d ' \n')
+
+		# check that the character count is not zero (this would indicate that there are status flags in this column)
+		if [ (count $column_status) -ne 0 ];
+			# we only want to display unique status flags (eg: if there are 5 modified files, the prompt should only show the modified status once)
+			set -l column_unique_status (echo $column_status | sort | uniq)
+			# parse the status flags for this column and create the formatting by calling out to the helper function
+			set -l svn_status_flags (__fish_svn_prompt_parse_status $column_unique_status)
+
+			# the default separator is empty
+			set -l prompt_separator ""
+			for index in (seq (math "$col - $last_column"))
```

Comment:
Since you're not using $index inside the loop, you can just do `seq $last_column $col`. Or am I missing anything?